lib/Geo/ShapeContainer.cpp

```diff
@@ -600,7 +651,10 @@ bool ShapeContainer::intersects(S2LatLngRect const* other) const {
     }
 
     case ShapeContainer::Type::S2_POLYLINE: {
-      return contains(other);
+      auto rectPoly = ::latLngRectToPolygon(static_cast<S2LatLngRect const*>(other));
```

Comment:
that makes absolutely no sense to allocate rectPoly on a heap

Reply:
Fixed.